spec/services/candidate_interface/decoupled_references/request_reference_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::DecoupledReferences::RequestReference do
+  describe '#call' do
```

Comment:
There's a lot of mocking here, but we really do care about the side effects. & we're tending towards double-entry testing where the spec basically knows how the class is implemented and a small change over there (or to, for example `ActionMailer`) can mean changes over here too. I'd not bother to mock the mailer, for example — would prefer to check an email's sent and that it was the right one.

Reply:
100% agree. I was following a pattern, but this pattern definitely tests implementation, not behaviour.